### ISSUE TITLE
[web] Flutter master branch breaks flutter_riverpod (only on release and profile mode)

### ISSUE THREAD

**Issue by borjandev** [rrousselGit/river_pod#416]
**Describe the bug**
Nasty bug which at the very least breaks StateNotifierProvider or more

- Happens only on Flutter Web
- Master branch only
- Release and profile mode only (debug mode works properly)
- Started happening from commit [https://github.com/flutter/flutter/commit/d226d439121d90302b3affbb56c5d3a7cc95d8d4](https://github.com/flutter/flutter/commit/d226d439121d90302b3affbb56c5d3a7cc95d8d4)
- Breaks all versions including flutter_riverpod 0.14.0

**To Reproduce**

```
import 'package:flutter/material.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';

void main() {
  runApp(const ProviderScope(child: MyApp()));
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(home: TrafficLight());
  }
}

class TrafficLight extends ConsumerWidget {
  @override
  Widget build(BuildContext context, ScopedReader watch) {
    final trafficLightColor = watch(colorChoiceProvider);
    switch (trafficLightColor) {
      case ColorChoiceModel.red:
        return Scaffold(
          appBar: AppBar(backgroundColor: Colors.red),
          floatingActionButton: FloatingActionButton(
            backgroundColor: Colors.red,
            child: Icon(Icons.traffic),
            onPressed: () {
              context.read(colorChoiceProvider.notifier).changeToYellow();
            },
          ),
        );

      case ColorChoiceModel.yellow:
        return Scaffold(
          appBar: AppBar(backgroundColor: Colors.yellow),
        );
      case ColorChoiceModel.green:
        return Scaffold(
          appBar: AppBar(backgroundColor: Colors.green),
        );
    }
  }
}

enum ColorChoiceModel {
  red,
  yellow,
  green,
}

final StateNotifierProvider<ColorChoice, ColorChoiceModel> colorChoiceProvider = StateNotifierProvider((ProviderReference ref) {
  return ColorChoice();
});

class ColorChoice extends StateNotifier<ColorChoiceModel> {
  ColorChoice() : super(ColorChoiceModel.red);

  void changeToRed() {
    state = ColorChoiceModel.red;
  }

  void changeToYellow() {
    state = ColorChoiceModel.yellow;
  }

  void changeToGreen() {
    state = ColorChoiceModel.green;
  }
}
```

```
flutter channel master
flutter upgrade
flutter doctor -v
[✓] Flutter (Channel master, 2.1.0-13.0.pre.471, on macOS 11.2.3 20D91 darwin-arm, locale en-MK)
    • Flutter version 2.1.0-13.0.pre.471 at /Users/dev/flutter
    • Framework revision bb7b490759 (85 minutes ago), 2021-04-06 03:00:10 -0400
    • Engine revision e74b6497b8
    • Dart version 2.13.0 (build 2.13.0-204.0.dev)
```

**Expected behavior**
When the FloatingActionButton with the traffic light icon is pressed the AppBar should change the color to yellow.

**Comment by rrousselGit** [comment]
Are you sure that this isn't the master channel that is broken?

**Comment by borjandev** [comment]
I suspected the same but no one seems to be raising any similar issues on the Flutter repository.

The commit below that "broke" things is already a few days old and there have been quite some additional commits afterwards. 

It wasn't easy to isolate [https://github.com/flutter/flutter/commit/d226d439121d90302b3affbb56c5d3a7cc95d8d4](https://github.com/flutter/flutter/commit/d226d439121d90302b3affbb56c5d3a7cc95d8d4) as the culprit.

I had to checkout the master branch of the Flutter repository commit by commit and re-download all of the SDK components on each checkout and rebuild the reproducible example I submitted again and again.

The default Flutter functionalities are working on the master branch, like pressing the FloatingActionButton etc..

I suspect it would be a bit difficult to isolate the root cause as it only appears on Web, and it only appears on Release and Profile mode.

Maybe an issue needs to be opened in the Flutter repository as well in order for them to fix it, if it's from their end, but it's tricky since flutter_riverpod is not fully broken, StateProvider works for example even on the master branch, so far I've encountered the issue on StateNotifierProvider specifically.

I was hoping you would have more ideas on what would be the best way to get it resolved?
